go/nvault/vault.go

```diff
@@ -193,34 +200,52 @@ func (v Vault) Lookup(key string) (string, error) {
 
 func (v Vault) Exists(key string) (bool, error) {
 	keyName := fmt.Sprintf("%s.key", key)
-	_, err := v.s3Client.HeadObject(context.TODO(), &s3.HeadObjectInput{Bucket: &v.cloudformationParams.BucketName, Key: &keyName})
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	_, err := v.s3Client.HeadObject(ctx, &s3.HeadObjectInput{
+		Bucket: &v.cloudformationParams.BucketName,
+		Key:    &keyName,
+	})
 	if err != nil {
-		var responseError *awshttp.ResponseError
-		if errors.As(err, &responseError) && responseError.ResponseError.HTTPStatusCode() == http.StatusNotFound {
+		var apiErr smithy.APIError
+		if errors.As(err, &apiErr) && apiErr.ErrorCode() == "NotFound" {
 			return false, nil
 		}
 		return false, err
 	}
+
 	return true, nil
 }
 
 func (v Vault) putS3Object(key string, value io.Reader, c chan error) {
-	_, err := v.s3Client.PutObject(context.TODO(), &s3.PutObjectInput{Bucket: &v.cloudformationParams.BucketName, Key: &key, Body: value})
-	if err != nil {
-		c <- err
-		return
-	}
-	c <- nil
+	defer close(c)
```

Comment:
Why are we closing a channel that we are given as argument? How come this doesn't break the tests, as if you look at the `Store`-func, this should cause the channel to be prematurely closed. Seems like a massive antipattern and this closing of the channel is already handled in the `Store`-func accordingly, so what's the point of this?
```suggestion
```